continuum/datasets/synbols.py

```diff
@@ -0,0 +1,237 @@
+            dataset_name (str): See https://github.com/ElementAI/synbols-resources/raw/master/datasets/generated/',
+            download (bool): Whether to download the dataset
+        """
+        if download:  # done here in order to pass x and y to super
+            full_path = get_data_path_or_download("default_n=100000_2020-Oct-19.h5py",
+                                                  data_root=data_path)
+        else:
+            full_path = os.path.join(data_path, dataset_name)
+
+        data = SynbolsHDF5(full_path,
+                           task_type,
+                           mask='random',
+                           trim_size=None,
+                           raw_labels=False)
+        data = SynbolsSplit(data, 'train' if train else 'val')
+
+        super().__init__(data.x, data.y, train=train, download=download)
+
+    @property
+    def transformations(self):
+        return [tt.ToPILImage(), tt.ToTensor()]
+
+
+def _read_json_key(args):
+    string, key = args
+    return json.loads(string)[key]
+
+
+class SynbolsHDF5:
+    """HDF5 Backend Class"""
+
+    def __init__(self, path, task, ratios=[0.6, 0.2, 0.2], mask=None, trim_size=None, raw_labels=False, reference_mask=None):
+        """Constructor: loads data and parses labels.
+
+        Args:
+            path (str): path where the data is stored (see full_path above)
+            task (str): 'char', 'font', or the field of choice 
+            ratios (list, optional): The train/val/test split ratios. Defaults to [0.6, 0.2, 0.2].
+            mask (ndarray, optional): Mask with the data partition. Defaults to None.
+            trim_size (int, optional): Trim the dataset to a smaller size, for debugging speed. Defaults to None.
+            raw_labels (bool, optional): Whether to include all the attributes of the synbols for each batch. Defaults to False.
+            reference_mask (ndarray, optional): If train and validation are done with two different datasets, the 
+                                                reference mask specifies the partition of the training data. Defaults to None.
+
+        Raises:
+            ValueError: Error message
+        """
+        self.path = path
+        self.task = task
+        self.ratios = ratios
+        print("Loading hdf5...")
+        with h5py.File(path, 'r') as data:
+            self.x = data['x'][...]
+            y = data['y'][...]
+            print("Converting json strings to labels...")
+            with multiprocessing.Pool(8) as pool:
+                self.y = pool.map(json.loads, y)
+            print("Done converting.")
+            if isinstance(mask, str):
+                if "split" in data:
+                    if mask in data['split'] and mask == "random":
+                        self.mask = data["split"][mask][...]
+                    else:
+                        self.mask = self.parse_mask(mask, ratios=ratios)
+                else:
+                    raise ValueError
```

Comment:
Can you provide a understandable error message in the exception. If it is raised, what should the user do?

Reply:
I simplified the whole thing and the error is no longer necessary

Reply:
Great thanks